stream2hop/tns2hop.py

```diff
@@ -0,0 +1,254 @@
+            ]
+
+        if "instrument" in spectra:
+            spectra["instrument"]["description"] = parameters_list["instrument"][
+                str(spectra["instrument"]["id"])
+            ]
+
+        if "telescope" in spectra:
+            spectra["telescope"]["description"] = parameters_list["telescope"][
+                str(spectra["telescope"]["id"])
+            ]
+
+        if "assoc_groups" in spectra:
+            for assoc_group in spectra["assoc_groups"]:
+                assoc_group["description"] = parameters_list["groups"][
+                    str(assoc_group["id"])
+                ]
+
+    return object
+
+
+def get_new_data(scimma_url, config, api_key):
+    """
+      Retrieve new TNS objects
+
+      Args:
+        scimma_url: scimma URL that will be used to publish data to it
+        config: configurations to access scimma URL
+
+      Returns:
+        None
+    """
+
+    url_tns_api = "https://wis-tns.weizmann.ac.il/api/get"
+
+    #  every 3 hours
+    days_ago = 0.125
+    date = str(datetime.utcnow() - timedelta(days=days_ago))
+    print("Running on: " + date)
+    #  Set search parameter
+    search_obj = {"public_timestamp": date}
+    response = search(url_tns_api, search_obj, api_key)
+
+    if None not in response:
+
+        #  New data is retrieved, open a stream with scimma
+        sC = ut.ScimmaConnection(scimma_url, config)
+        sC.open()
+
+        #  Read parameters file
+        parameters_list = json.load(open("../parameters.json", "r"))["data"]
+        objects_list = json.loads(response.text)["data"]["reply"]
+
+        for object in objects_list:
+            get_obj = {
+                "objname": object["objname"],
+                "photometry": "1",
+                "spectra": "1",
+                "classification": "1",
+                "discovery": "1",
+                "AT": "1",
+            }
+            response = get_object(url_tns_api, get_obj, api_key)
+
+            if response:
+                object_data = json.loads(response.text)["data"]["reply"]
+                object_data = fix_photometery(object_data, parameters_list)
+                object_data = fix_spectra(object_data, parameters_list)
+                object_data = {"format": "BLOB", "content": object_data}
+                sC.write(json.dumps(object_data, indent=4))
+        sC.close()
+    else:
+        print("Nothing to do\n")
+
+    pass
+
+
+def get_astronotes(scimma_url, config, api_key):
+    """
+        Get the latest astronotes in the previous day
+
+        Args:
+            scimma_url: scimma URL that will be used to publish data to it
+            config: configurations to access scimma URL
+
+        Returns:
+            None
+    """
+    date = datetime.today().strftime("%Y-%m-%d")
+    astronotes_url = (
+        "https://wis-tns.weizmann.ac.il/astronotes?&date_start%5Bdate%5D="
+        + date
+        + "&format=csv"
+    )
+    wget.download(astronotes_url, "astronotes_today.csv")
```

Comment:
Is there a reason you used both `wget` and `requests`? It might be better to use `requests` only and keep the dependencies down unless there's a critical feature you need.

For example, I think you may be able to get what you want by calling `requests.get()` directly and parsing the csv file, rather than dumping it to disk and then re-reading it. You should be able to do the same steps as below by feeding the output to an `io.StringIO()` instance.

Reply:
No particular reason actually. I will change it to requests.get
Thanks